crates/assistant_tools/src/path_search_tool.rs

```diff
@@ -47,42 +47,44 @@ impl Tool for PathSearchTool {
         project: Entity<Project>,
         cx: &mut App,
     ) -> Task<Result<String>> {
-        let glob = match serde_json::from_value::<PathSearchToolInput>(input) {
-            Ok(input) => input.glob,
-            Err(err) => return Task::ready(Err(anyhow!(err))),
-        };
-        let path_matcher = match PathMatcher::new(&[glob.clone()]) {
-            Ok(matcher) => matcher,
-            Err(err) => return Task::ready(Err(anyhow!("Invalid glob: {}", err))),
-        };
+        let worktrees: Vec<_> = project.read(cx).worktrees(cx).collect();
 
-        let mut matches = Vec::new();
```

Comment:
This is still spawning on the main thread. To spawn on a background thread, you should use `cx.background_spawn`.

Reply:
@as-cii Should be good now!

Reply:
@as-cii FYI, this is now ready to go, only blocked on your approval (if it looks good now).